sys/dev/cxgbe/iw_cxgbe/cq.c

```diff
@@ -132,9 +132,12 @@ create_cq(struct c4iw_rdev *rdev, struct t4_cq *cq,
 	wr_len = sizeof *res_wr + sizeof *res;
 
 	wr = alloc_wrqe(wr_len, &sc->sge.ctrlq[0]);
-	if (wr == NULL)
-		return (0);
-        res_wr = wrtod(wr);
+	if (!wr) {
```

Comment:
Not an expert on this code but it seems like there should be calls to `free_wrqe()` (and might need an `err5` label). Not seeing this in either `create_cq()` or `destroy_cq()`.

Reply:
I'm aware. However I want to investigate and make sure this memory is being leaked because a double free is arguably worse.